# NHA-hTERT


### FLAIR

```sh 
# Define the variables:
genome=/mnt/ref/GCA_000001405.15_GRCh38_no_alt_analysis_set.fa
transcriptome_annotation=/mnt/ref/gencode.v38.annotation.gtf
minimap2=/minimap2-2.14_x64-linux/minimap2
flair=/scratch/flair/flair.py
sample=NHA_hTERT_DRNA_20220609
sample_output=${sample} 
data_dir=/mnt/flair_data/${sample}/
fastq=${data_dir}/NHA-hTERT_DRNA_20220609_Guppy_3.1.5_pass.fastq.gz 
THid=TH46_2685_S01
illumina_bam=/mnt/flair_data/${THid}/${THid}_illumina.sortedByCoord.md.bam 
cd $data_dir



# FLAIR-align command:
python3 /scratch/flair/flair.py align -g $genome -r $fastq -o ${sample}_aligned_flair_output -m $minimap2 --version1.3 --nvrna

# Extract splice junctions from short-read alignments:
python3 /scratch/flair/bin/junctions_from_sam.py -s $illumina_bam -n ${THid}_illumina_flair_splice_junctions.bed

# Remove infrequently used junctions
# It is recommended to remove infrequently used junctions. Filter junctions with fewer than 3 supporting short reads out:
awk '{ if ($5 > 2) { print } } ' ${THid}_illumina_flair_splice_junctions.bed_junctions.bed > ${THid}_illumina_flair_splice_junctions_filtered.bed

# FLAIR-correct command:
python3 $flair correct -q ${sample}_aligned_flair_output.bed -g $genome -o ${sample}_flair_correct_output -f $transcriptome_annotation -j ${THid}_illumina_flair_splice_junctions_filtered.bed -c /mnt/ref/GCA_000001405.15_GRCh38_no_alt_analysis_set.chrom_sizes

# FLAIR-collapse command:
python3 $flair collapse -g $genome -r $fastq -q ${sample}_flair_correct_output_all_corrected.bed -f $transcriptome_annotation -m $minimap2 -o ${sample}_flair_collapse_output
```

#### Align fastq to custom transcripts generated by FLAIR

```sh 
# Define the variables:
custom_genome=${data_dir}/${sample}_flair_collapse_output.isoforms.fa

# FLAIR-align to transcriptome
python3 ${flair} align -g $custom_genome -r $fastq -o ${sample}_self_transcript_aligned -m $minimap2 --version1.3 --nvrna
```