# DIPG-IV combined runs


### FLAIR

```sh 
# Define the variables:
genome=/mnt/ref/GCA_000001405.15_GRCh38_no_alt_analysis_set.fa
transcriptome_annotation=/mnt/ref/gencode.v38.annotation.gtf
minimap2=/minimap2-2.14_x64-linux/minimap2
flair=/mnt/flair/flair.py
sample=DIPG_IV_DRNA_20220314_and_20220401
sample_output=DIPG_IV_DRNA_20220314_and_20220401_v2
data_dir=/mnt/flair_data/${sample}/
fastq=${data_dir}/DIPG-IV_DRNA_20220314+20220401_combined_Guppy_3.1.5_pass.fastq.gz
cell_line=DIPG-IV
illumina_bam=/mnt/flair_data/THR13_0970_S11/THR13_0970_S11_illumina.sortedByCoord.md.bam


# FLAIR-align command:
python3 /mnt/flair/flair.py align -g $genome -r $fastq -o ${sample}_aligned_flair_output -m $minimap2 --version1.3 --nvrna

# Extract splice junctions from short-read alignments:
python3 /mnt/flair/bin/junctions_from_sam.py -s $illumina_bam -n ${cell_line}_illumina_flair_splice_junctions.bed

# Remove infrequently used junctions
# It is recommended to remove infrequently used junctions. Filter junctions with fewer than 3 supporting short reads out:
awk '{ if ($5 > 2) { print } } ' ${cell_line}_illumina_flair_splice_junctions.bed_junctions.bed > ${cell_line}_illumina_flair_splice_junctions_filtered.bed

# FLAIR-correct command:
python3 $flair correct -q ${sample}_aligned_flair_output.bed -g $genome -o ${sample}_flair_correct_output -f $transcriptome_annotation -j ${cell_line}_illumina_flair_splice_junctions_filtered.bed -c /mnt/ref/GCA_000001405.15_GRCh38_no_alt_analysis_set.chrom_sizes

# FLAIR-collapse command:
python3 $flair collapse -g $genome -r $fastq -q ${sample}_flair_correct_output_all_corrected.bed -f $transcriptome_annotation -m $minimap2 -o ${sample}_flair_collapse_output
```

#### Align fastq to custom transcripts generated by FLAIR

```sh 
# Define the variables:
source_dir=/mnt/flair_data/DIPG_IV_DRNA_20220314_and_20220401
genome=${source_dir}/flair_output/DIPG_IV_DRNA_20220314_and_20220401_flair_collapse_output.isoforms.fa
fastq=${source_dir}/DIPG-IV_DRNA_20220314+20220401_combined_Guppy_3.1.5_pass.fastq.gz
analysis_name=full_fq_to_sample_transcripts
out_dir=${source_dir}/${analysis_name}
mkdir $out_dir
minimap2=/minimap2-2.14_x64-linux/minimap2

# FLAIR-align to transcriptome
python3 /scratch/flair/flair.py align -g $genome -r $fastq -o ${analysis_name}_output -m $minimap2 --version1.3 --nvrna
```